Core/clim-basic/ports.lisp

```diff
@@ -168,12 +169,96 @@
   (with-slots (properties) port
     (setf (getf properties indicator) value)))
 
+(defmethod distribute-event ((port basic-port) event)
+  (dispatch-event (event-sheet event) event))
+
+;;; This method is responsible for synthetising leave/enter events for
+;;; sheets which are not mirrored. -- jd 2019-08-05
+(defmethod distribute-event :around ((port basic-port) (event pointer-event))
```

Comment:
I'm afraid that this might turn out to be inflexible. What if a new port has its own way of generating enter and exit events?

Reply:
This method is responsible for fixing the port-pointer sheet and none of unmirrored sheets will receive its events without such tracking. This code was around for some time and it is basically how we do things right now.

I have two answers for that:
- If some port has different way, then it should subclass port, not basic-port
- We'll worry about a situation when the problem occur (and maybe we'll change design)

While CLX in a hacky manner sneaks mirror for each sheet (via frame-manager) it is abnormal, in ordinary case it is the programmer who makes their class have a mirror (by mixing a specific class).

Reply:
OK.